test/new-e2e/tests/agent-platform/common/test_client.go

```diff
@@ -184,7 +185,7 @@ func (c *TestClient) ExecuteWithRetry(cmd string) (string, error) {
 		if err == nil {
 			ok = true
```

Comment:
not strictly related to your PR, but it would probably be better to `break` or even directly `return` here, as even in the case of success we'll still sleep afterward, while there's no need for it

Reply:
🙈